### ISSUE TITLE
Proposal for Adding New Examples to mlpack's Example Zoo

### ISSUE THREAD

**Issue by sitamgithub-MSIT** [mlpack/examples#220]
## Objective
The objective of this proposal is to enhance mlpack's example repository by adding new examples that demonstrate the application of various machine learning algorithms across diverse real-world domains, as explained [here](https://github.com/mlpack/mlpack/wiki/SummerOfCodeIdeas#example-zoo).

## Proposed Deliverables
This aims to add new examples to mlpack's repository, provide detailed documentation, integrate visualizations, and ensure regular updates and maintenance to ensure compatibility with the latest versions of mlpack and related dependencies.

## Example Ideas
These examples demonstrate the potential of mlpack in various tasks, such as image recognition and others.

cc: @zoq

**Comment by sitamgithub-MSIT** [comment]
I went through the mentioned tickets. What I got from those was: 

#154 In this issue, it is proposed to put examples in their respective folders, like this:

```
lstm_electricity_consumption/
|- .
|- ..
|- Jupyter_notebook_code/   <-- code relative to Jupyter notebook
|- C++/    <-- code relative to C++, Makefiles, etc.
|- Go/     <-- similar thing here
|- Python/
|- etc/   <-- other language binding 
README.md <-- this will explain what is happening inside this folder.
```
One pull request already mentioned with respect to this issue [here](https://github.com/mlpack/examples/pull/191)

#113 Here it is proposed to give an example using YOLOv1, and the issue says to first implement YOLOv3 and PR get merged, and then we can make an example after that.

**Comment by sitamgithub-MSIT** [comment]
Though I want to go ahead with making examples and, if required, also helping in implementing YOLOv3 later.

**Comment by zoq** [comment]
https://github.com/mlpack/examples/issues/154 as you already mentioned is about restructuring the repository, and is more about the build pipeline to make sure each example builds as expected, https://github.com/mlpack/examples/pull/191 is a very good basis.

**Comment by zoq** [comment]
We can provide an example for YOLOv1, no need to wait for an implementation of v3, but nevertheless, would be great to have for sure.

**Comment by sitamgithub-MSIT** [comment]
For YOLOv1, I had read the issue, but I would like to spend a little time researching and then coming up with my proposal. For the file structure referring to issue #191, hopefully we will raise an initial PR soon.

cc: @zoq

**Comment by sitamgithub-MSIT** [comment]
> Sounds good, let me know if there is anything that needs clarification.

Currently, I am referring to PR #191 for my initial work. Moving different files or notebooks to their corresponding folders or updating the readme should not be a problem. Handling the makefile and others is something I am particularly interested in. Btw, these are the initial things I observed when revising the PR, so I can be wrong as well. After a thorough review of the PR, things will be more clear.

**Comment by sitamgithub-MSIT** [comment]
Btw @zoq this should be the folder structure right!?
```
lstm_electricity_consumption/
|- .
|- ..
|- Jupyter_notebook_code/   <-- code relative to Jupyter notebook
|- C++/    <-- code relative to C++, Makefiles, etc.
|- Go/     <-- similar thing here
|- Python/7
|- etc/   <-- other language binding 
README.md <-- this will explain what is happening inside this folder.
```

**Comment by sitamgithub-MSIT** [comment]
Btw @zoq this should be the folder structure right!?
```
lstm_electricity_consumption/
|- .
|- ..
|- Jupyter_notebook_code/   <-- code relative to Jupyter notebook
|- C++/    <-- code relative to C++, Makefiles, etc.
|- Go/     <-- similar thing here
|- Python/7
|- etc/   <-- other language binding 
README.md <-- this will explain what is happening inside this folder.
```

**Comment by shrit** [comment]
@sitamgithub-MSIT keep in mind that all of the current examples are pure Jupyter Notebook which is only handy when needed. I think we should export C++ examples for each one of these examples.
Originally this repo used to have C++ examples, now it looks more like Jupyter Notebook examples, it would be nice to have some pure C++ examples as well.

**Comment by shrit** [comment]
`etc/` does not make much sense, if it is a Julia example, then you put it in a Julia directory. There is no need to have exactly the same directory in each one of the examples.

**Comment by Shay2Shay** [comment]
I wanted to contribute into Example Zoo. I just wanted to know does MlPack provide Learnable Parameters as Pytorch do?
(Learnable Parameter = any tensor / matrix / array which can be adjusted via gradient descent).
